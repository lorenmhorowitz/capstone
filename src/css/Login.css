.Img {
  background-image: url("https://cdn.discordapp.com/attachments/489154865344282625/508368921666256936/Office.png");
  background-position: center;
  background-repeat: no-repeat;
  background-size: cover;
  top: 0%;
  bottom: 0%;
  right: 0%;
  left: 0%;
  position: fixed;
  overflow-y: auto;
}

.loginSquare {
  background-color: rgba(38, 38, 38, 0.8);
  border: 2px solid #f96302;
  width: 35vw;
  margin-left: 10vw;
  margin-top: 12vw;
  min-width: 350px;
  position: relative;
  text-align: center;
}

.grid-container {
  display: grid;
  grid-template-columns: 15vw auto;
  max-width: 100%;
  grid-gap: 0%;
  padding: 0%;
  margin: 0%;
  justify-content: space-evenly;
}

.grid-item {
  padding: 0;
  text-align: center;
}

#title {
  margin-top: 3%;
  text-align: center;
  font-size: 4em;
}

#Corner {
  color: #ffffff;
  font-family: "Cutive", serif;
  font-weight: bolder;
}

#Stone {
  color: #f96302;
  font-family: "Cutive", serif;
  font-weight: bolder;
}

#ButtonWithoutBottomMargin {
  background-color: #898989;
  border-radius: 8px;
  color: #ffffff;
  font-size: 1.2em;
  height: 60px;
  margin-top: 5%;
  width: 75%;
}

#ButtonWithBottomMargin {
  background-color: #898989;
  border-radius: 8px;
  color: #ffffff;
  font-size: 1.2em;
  height: 60px;
  margin-top: 5%;
  margin-bottom: 5%;
  width: 75%;
}

#HoverLogin {
  background-color: #26ab7a;
  border-radius: 8px;
  color: #ffffff;
  font-size: 1.2em;
  height: 60px;
  margin-top: 5%;
  margin-bottom: 5%;
  width: 75%;
}

#error {
  margin-top: 15px;
  margin-left: 15%;
  margin-right: 15%;
  margin-bottom: 5px;
  display: inline-block;
  color: white;
  font-size: 18px;
}

@media only screen and (min-width: 501px) and (max-width: 1100px) {
  .loginSquare {
    margin-right: 0%;
    margin-left: 0%;
    align-content: center;
  }

  #title {
    font-size: 3.5em;
  }
}

@media only screen and (min-width: 0px) and (max-width: 500px) {
  .grid-container {
    grid-template-columns: auto auto;
    justify-content: center;
  }

  .loginSquare {
    margin-right: 0%;
    margin-left: 0%;
    align-content: center;
  }

  #title {
    font-size: 3em;
  }
}

@media only screen and (min-width: 2000px) {
  .loginSquare {
    height: 70%;
  }
}
