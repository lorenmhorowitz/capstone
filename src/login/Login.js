import React, { Component } from "react";
import "../css/Login.css";
import Button from "@material-ui/core/Button";
import request from "request";
import { Redirect } from "react-router-dom";
import TextField from "../components/TextField";

const LOGIN_ACCOUNT_API =
  "https://us-central1-hdqc-capstone.cloudfunctions.net/login";

class Login extends Component {
  constructor(props) {
    super(props);
    this.state = {
      loginError: false,
      password: "",
      redirect: false,
      username: "",
      home: false
    };
  }

  handleEmailChange = data => {
    this.setState({ username: data.target.value });
  };

  handlePasswordChange = data => {
    this.setState({ password: data.target.value });
  };

  handleHoverLogin = () => {
    this.setState({ redirect: true });
  };

  handleLogin = () => {
    request.post(
      {
        url: LOGIN_ACCOUNT_API,
        headers: {
          "Content-Type": "application/json"
        },
        json: true,
        body: {
          kind: "HoverAuthentication",
          key: this.state.username,
          value: {
            password: this.state.password
          }
        }
      },
      function(error, response) {
        if (error) {
          return;
        }

        if (response.statusCode === 200) {
          this.setState({
            loginError: false,
            redirect: true,
            home: true
          });
        } else {
          this.setState({ loginError: true });
        }
      }.bind(this)
    );
  };

  render() {
    if (this.state.redirect && this.state.home === false) {
      return <Redirect push to="/hover" />;
    } else if (this.state.redirect && this.state.home) {
      return <Redirect push to="/home" />;
    }

    const errorMessage = <p id="error">Unable to log in. Please try again.</p>;

    return (
      <div className="Img" id="scroll">
        <div id="loginSquare">
          <div id="title">
            <span id="Corner">Corner</span>
            <span id="Stone">Stone</span>
          </div>
          <TextField label="email" handler={this.handleEmailChange} />
          <TextField label="password" handler={this.handlePasswordChange} />
          <Button id="Login" onClick={this.handleLogin}>
            Log In
          </Button>

          <Button id="HoverLogin" onClick={this.handleHoverLogin}>
            Login With Hover
          </Button>
          {this.state.loginError ? errorMessage : null}
        </div>
      </div>
    );
  }
}

export default Login;
