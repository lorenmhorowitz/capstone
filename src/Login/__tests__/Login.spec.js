import React from "react";
import Login from "../Login";
import { shallow } from "enzyme";

describe("The login page renders properly", () => {
  it("Login component renders the login page properly", () => {
    let wrapper = shallow(<Login />);
    expect(wrapper).toMatchSnapshot();
  });
});
