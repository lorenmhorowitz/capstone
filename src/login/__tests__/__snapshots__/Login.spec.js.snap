// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The login page renders properly Login component renders the login page properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Login />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        id="loginSquare"
      >
        <div
          id="title"
        >
          <span
            id="Corner"
          >
            Corner
          </span>
          <span
            id="Stone"
          >
            Stone
          </span>
        </div>
        <TextField
          handler={[Function]}
          label="email"
        />
        <TextField
          handler={[Function]}
          label="password"
        />
        <WithStyles(Button)
          id="Login"
          onClick={[Function]}
        >
          Log In
        </WithStyles(Button)>
        <WithStyles(Button)
          id="HoverLogin"
          onClick={[Function]}
        >
          Login With Hover
        </WithStyles(Button)>
      </div>,
      "className": "Img",
      "id": "scroll",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            id="title"
          >
            <span
              id="Corner"
            >
              Corner
            </span>
            <span
              id="Stone"
            >
              Stone
            </span>
          </div>,
          <TextField
            handler={[Function]}
            label="email"
          />,
          <TextField
            handler={[Function]}
            label="password"
          />,
          <WithStyles(Button)
            id="Login"
            onClick={[Function]}
          >
            Log In
          </WithStyles(Button)>,
          <WithStyles(Button)
            id="HoverLogin"
            onClick={[Function]}
          >
            Login With Hover
          </WithStyles(Button)>,
          null,
        ],
        "id": "loginSquare",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                id="Corner"
              >
                Corner
              </span>,
              <span
                id="Stone"
              >
                Stone
              </span>,
            ],
            "id": "title",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Corner",
                "id": "Corner",
              },
              "ref": null,
              "rendered": "Corner",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Stone",
                "id": "Stone",
              },
              "ref": null,
              "rendered": "Stone",
              "type": "span",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "handler": [Function],
            "label": "email",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "handler": [Function],
            "label": "password",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Log In",
            "id": "Login",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Log In",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Login With Hover",
            "id": "HoverLogin",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Login With Hover",
          "type": [Function],
        },
        null,
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          id="loginSquare"
        >
          <div
            id="title"
          >
            <span
              id="Corner"
            >
              Corner
            </span>
            <span
              id="Stone"
            >
              Stone
            </span>
          </div>
          <TextField
            handler={[Function]}
            label="email"
          />
          <TextField
            handler={[Function]}
            label="password"
          />
          <WithStyles(Button)
            id="Login"
            onClick={[Function]}
          >
            Log In
          </WithStyles(Button)>
          <WithStyles(Button)
            id="HoverLogin"
            onClick={[Function]}
          >
            Login With Hover
          </WithStyles(Button)>
        </div>,
        "className": "Img",
        "id": "scroll",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              id="title"
            >
              <span
                id="Corner"
              >
                Corner
              </span>
              <span
                id="Stone"
              >
                Stone
              </span>
            </div>,
            <TextField
              handler={[Function]}
              label="email"
            />,
            <TextField
              handler={[Function]}
              label="password"
            />,
            <WithStyles(Button)
              id="Login"
              onClick={[Function]}
            >
              Log In
            </WithStyles(Button)>,
            <WithStyles(Button)
              id="HoverLogin"
              onClick={[Function]}
            >
              Login With Hover
            </WithStyles(Button)>,
            null,
          ],
          "id": "loginSquare",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span
                  id="Corner"
                >
                  Corner
                </span>,
                <span
                  id="Stone"
                >
                  Stone
                </span>,
              ],
              "id": "title",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Corner",
                  "id": "Corner",
                },
                "ref": null,
                "rendered": "Corner",
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Stone",
                  "id": "Stone",
                },
                "ref": null,
                "rendered": "Stone",
                "type": "span",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "handler": [Function],
              "label": "email",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "handler": [Function],
              "label": "password",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Log In",
              "id": "Login",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Log In",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Login With Hover",
              "id": "HoverLogin",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Login With Hover",
            "type": [Function],
          },
          null,
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
